src/priceClient/priceClient.ts

```diff
@@ -0,0 +1,179 @@
+      if (age > this.maxPriceAge) {
+        missed[address] = age;
+      }
+    });
+
+    if (Object.keys(missed).length > 0) {
+      const requestAddresses = Object.keys(priceCache);
+      this.logger.debug({
+        at: "PriceClient#getPricesByAddress",
+        message: `${currency.toUpperCase()} cache miss (age > ${this.maxPriceAge} S).`,
+        tokens: missed,
+      });
+      const prices: PriceCache = await this.requestPrices(requestAddresses, currency);
+      if (Object.keys(prices).length === 0) {
+        this.logger.warn({ at: "PriceClient#getPricesByAddress", message: "Failed to update token prices." });
+        // @todo throw ?
+        return [];
+      }
+      this.updateCache(priceCache, prices, requestAddresses);
+    }
+
+    return addresses.map((addr: string) => {
+      const { price, timestamp } = priceCache[addr.toLowerCase()];
+      return { address: addr, price, timestamp };
+    });
+  }
+
+  expireCache(currency: string): void {
+    const priceCache = this.getPriceCache(currency);
+    Object.values(priceCache).forEach((token: TokenPrice) => (token.timestamp = 0));
+    this.logger.debug({ at: "PriceClient#expireCache", message: `Expired ${currency} cache.` });
+  }
+
+  protected getPriceCache(currency: string): PriceCache {
+    if (this.prices[currency] === undefined) this.prices[currency] = {};
+    return this.prices[currency];
+  }
+
+  private async requestPrices(addresses: string[], currency: string): Promise<PriceCache> {
+    let prices: TokenPrice[] = [];
+
+    for (const priceFeed of this.priceFeeds) {
+      this.logger.debug({
+        at: "PriceClient#getPriceByAddress",
+        message: `Looking up prices via ${priceFeed.name}.`,
+      });
+      try {
+        prices = await priceFeed.getPricesByAddress(addresses, currency);
+        if (prices.length === 0) {
+          throw Error("Zero-length response received");
+        }
+      } catch (err) {
+        this.logger.debug({
+          at: "PriceClient#requestPrices",
+          message: `Price lookup against ${priceFeed.name} failed (${err}).`,
+          tokens: addresses,
+        });
+        continue; // Failover to the next price feed...
+      }
+    }
+
+    if (prices.length === 0) {
+      throw Error(`Price lookup failed against all price feeds (${this.listPriceFeeds().toString()})`);
+    }
+
+    return Object.fromEntries(prices.map((price) => [price.address.toLowerCase(), price]));
+  }
+
+  private updateCache(priceCache: PriceCache, prices: PriceCache, expected: string[]): void {
+    const updated: TokenPrice[] = [];
+    const skipped: { [token: string]: string } = {}; // Includes reason for skipping
+
+    expected.forEach((address: string) => {
+      const addr = address.toLowerCase(); // for internal priceCache lookup.
+      const tokenPrice: TokenPrice | undefined = prices[addr];
+      const now: number = msToS(Date.now());
```

Comment:
This is probably dumb, since there's not likely to be any meaningful time difference between loops.

Reply:
Fixed: https://github.com/across-protocol/sdk-v2/pull/98/commits/53b8f48bd5af77693d1ea4e1a4655d627d1ee36d